bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gpi/GpiCB.java

```diff
@@ -44,46 +44,61 @@ protected void thisReset(Map<String, Object> options) {
     }
 
-    @Override
     protected boolean thisHasConfiguration() {
         if (_sysConfig == null)
             return false;
         return (_sysConfig.getParameterCount() > 0);
     }
 
-    @Override
-    protected void thisApplyNext(final IConfig config, final IConfig prevFull) {
+    protected void thisApplyNext(final IConfig config, IConfig prevFull) {
         final String systemName = _sysConfig.getSystemName();
-        final Collection<IParameter> sysConfig = _sysConfig.getParameters();
+        Collection<IParameter> sysConfig = _sysConfig.getParameters();
+
+        final class MagnitudeFilter implements Function1<Magnitude, Boolean> {
+            private final MagnitudeBand band;
+
+            private MagnitudeFilter(MagnitudeBand band) {
+                this.band = band;
+            }
+
+            @Override
+            public Boolean apply(Magnitude magnitude) {
+                return magnitude.band().equals(band);
+            }
+        }
+
+        final class MagnitudeSetter implements ApplyOp<Magnitude> {
+            private final String magProp;
+
+            private MagnitudeSetter(String magHProp) {
+                this.magProp = magHProp;
+            }
 
-        for (final IParameter param : sysConfig) {
+            @Override
+            public void apply(Magnitude magnitude) {
+                config.putParameter(systemName,
+                        DefaultParameter.getInstance(magProp,
+                                magnitude.value()));
+            }
+        }
+        for (IParameter param : sysConfig) {
             config.putParameter(systemName,
                     DefaultParameter.getInstance(param.getName(), param.getValue()));
         }
         config.putParameter(systemName,
                 StringParameter.getInstance(InstConstants.INSTRUMENT_NAME_PROP,
                         Gpi.INSTRUMENT_NAME_PROP));
-
-        // Set the magnitude properties.
         final ISPObsComponent targetcomp = SPTreeUtil.findTargetEnvNode(getObsComponent().getContextObservation());
         if (targetcomp != null) {
             final TargetObsComp toc = (TargetObsComp) targetcomp.getDataObject();
             if (toc != null) {
-                // The Asterism here should be Single(base), so we only need to concern ourselves with the head target.
-                toc.getAsterism().allSpTargetsJava().headOption().foreach(base -> {
-                    final ImList<Magnitude> magnitudes = base.getMagnitudesJava();
-                    MAG_PROPERTIES.forEach((band, propName) ->
-                        magnitudes.find(m -> m.band().equals(band)).
-                                foreach(mag -> config.putParameter(systemName, DefaultParameter.getInstance(propName, mag.value()))));
-                });
+                if (toc.getAsterism().allSpTargetsJava().head() != null) {
+                    ImList<Magnitude> magnitudes = toc.getAsterism().allSpTargetsJava().head().getMagnitudesJava();
```

Comment:
This line and the preceding one needed to be changed for the new way targets are handled, but everything else is as it was back beforehand.